aws/data_source_aws_organizations_delegated_administrators.go

```diff
@@ -0,0 +1,120 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/organizations"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceAwsOrganizationsDelegatedAdministrators() *schema.Resource {
+	return &schema.Resource{
+		ReadWithoutTimeout: dataSourceAwsOrganizationsDelegatedAdministratorsRead,
+		Schema: map[string]*schema.Schema{
+			"service_principal": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+			"delegated_administrators": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"arn": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"delegation_enabled_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"email": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"joined_method": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"joined_timestamp": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAwsOrganizationsDelegatedAdministratorsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).organizationsconn
+
+	input := &organizations.ListDelegatedAdministratorsInput{}
+
+	if v, ok := d.GetOk("service_principal"); ok {
+		input.ServicePrincipal = aws.String(v.(string))
+	}
+
+	var delegators []*organizations.DelegatedAdministrator
+
+	err := conn.ListDelegatedAdministratorsPagesWithContext(ctx, input, func(page *organizations.ListDelegatedAdministratorsOutput, lastPage bool) bool {
+		if page == nil {
+			return !lastPage
+		}
+
+		delegators = page.DelegatedAdministrators
```

Comment:
In case there are multiple pages returned, we should collect results over multiple pages

```suggestion
		delegators = append(delegators, page.DelegatedAdministrators...)
```